### ISSUE TITLE
require('jsdom') throws error

### ISSUE THREAD

**Issue by techkey** [tmpvar/jsdom#1220]
Hi all,

I thought to give `jsdom` a try.

First I've installed `io.js` using `iojs-v3.2.0-x64.msi` from `iojs.org`. I've installed it in the `d:\iojs` directory.

So far so good, then I installed `jsdom` with the command line `npm install jsdom`:

```
jsdom@6.2.0 node_modules\jsdom                                                                                                                        
├── xml-name-validator@2.0.1                                                                                                                          
├── xtend@4.0.0                                                                                                                                       
├── xmlhttprequest@1.7.0                                                                                                                              
├── browser-request@0.3.3                                                                                                                             
├── symbol-tree@3.1.2                                                                                                                                 
├── cssom@0.3.0                                                                                                                                       
├── nwmatcher@1.3.6                                                                                                                                   
├── parse5@1.5.0                                                                                                                                      
├── tough-cookie@1.2.0                                                                                                                                
├── acorn@1.2.2                                                                                                                                       
├── acorn-globals@1.0.5 (acorn@2.3.0)                                                                                                                 
├── cssstyle@0.2.29                                                                                                                                   
├── whatwg-url-compat@0.6.4 (tr46@0.0.2)                                                                                                              
├── escodegen@1.6.1 (estraverse@1.9.3, esutils@1.1.6, esprima@1.2.5, source-map@0.1.43, optionator@0.5.0)                                             
├── htmlparser2@3.8.3 (domelementtype@1.3.0, entities@1.0.0, domhandler@2.3.0, readable-stream@1.1.13, domutils@1.5.1)                                
└── request@2.61.0 (aws-sign2@0.5.0, stringstream@0.0.4, forever-agent@0.6.1, caseless@0.11.0, tunnel-agent@0.4.1, oauth-sign@0.8.0, isstream@0.1.2, j
son-stringify-safe@5.0.1, extend@3.0.0, qs@4.0.0, node-uuid@1.4.3, form-data@1.0.0-rc3, mime-types@2.1.5, http-signature@0.11.0, bl@1.0.0, combined-st
ream@1.0.5, hawk@3.1.0, har-validator@1.8.0)                                                                                                          
```

No errors so far, then I wrote a very basic test file, `test.js` containing:

```javascript
var jsdom;

jsdom = require('jsdom');
```

And that immediately throws a error:

```
> iojs.exe test.js                                                                                                      
d:\iojs\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:12                                                      
  var baseElements = document.getElementsByTagName("base");                                                             
                             ^                                                                                          
                                                                                                                        
TypeError: Cannot read property 'getElementsByTagName' of undefined                                                     
    at getDocumentBaseUrl (d:\iojs\node_modules\jsdom\lib\jsdom\browser\resource-loader.js:12:30)                       
    at Location.getTheBase (d:\iojs\node_modules\jsdom\lib\jsdom\browser\location.js:26:12)                             
    at setTheInput (d:\iojs\node_modules\jsdom\node_modules\whatwg-url-compat\lib\url.js:1116:78)                       
    at Object.module.exports.mixinURLUtils (d:\iojs\node_modules\jsdom\node_modules\whatwg-url-compat\lib\url.js:1429:3)
    at Object.<anonymous> (d:\iojs\node_modules\jsdom\lib\jsdom\browser\location.js:23:11)                              
    at Module._compile (module.js:430:26)                                                                               
    at Object.Module._extensions..js (module.js:448:10)                                                                 
    at Module.load (module.js:355:32)                                                                                   
    at Function.Module._load (module.js:310:12)                                                                         
    at Module.require (module.js:365:17)                                                                                
``` 

The used OS is `Windows 10 Pro x64`.

Do I miss something from the basic examples in the `README.md`?

Any help would be much appreciated!

Cornelis.

**Comment by Joris-van-der-Wel** [comment]
This error did not occur for me until I updated `whatwg-url-compat` @sebmaster

`mixinURLUtils()` (whatwg url.js) calls `getTheBase()` (jsdom browser/location.js), without creating a `Location` so there is no document

**Comment by mathieuancelin** [comment]
Yep, I have the same issue and it's definitely an issue with last `whatwg-url-compat` version

**Comment by Sebmaster** [comment]
Fixed with whatwg-url-compat 0.6.5.

**Comment by techkey** [comment]
After updating `jsdom` it works as expected, thanks a lot!
